YAFCparser/Data/FactorioDataDeserializer_Context.cs

```diff
@@ -390,7 +390,10 @@ public void SealAndDeduplicate(TValue[] addExtra = null, IComparer<TValue> sort
                         storage[key] = prev;
                     else
                     {
-                        var mergedList = addExtra == null ? list : list.Concat(addExtra); 
+                        // Filter the extra items based on the current key, value being processed
+                        var tempExtra = filter != null ? addExtra.Where(filter(key)) : addExtra;
```

Comment:
Related to the comment by @SWeini: This will break (throw some exception) when `filter != null` and `addExtra == null`. Which seems to be possible as the original code checked if `addExtra == null`.